### ISSUE TITLE
On-the-fly Connections

### ISSUE THREAD

**Issue by OJ** [basho/riak-dotnet-client#111]
From the Riak Mailing list:

TL;DR version: is there a way to use Corrugated Iron making connections on
the fly rather than a fixed connection pool?

Explanation:

We are using Corrugated Iron on several (5) components that talk to Riak.
We want to be able to independently scale the number of instances of these
components live in production and we're having problems understanding how
best to do this with the connection pool assigned in CI.

Our Riak instance (currently 5 nodes) is load balanced behind an HAProxy
instance and we have been using the RiakExternalLoadBalancer class pointed
to this load balancer to instantiate our RiakClient. HAProxy is configured
at the moment to permit up to 250 connections to each Riak node (we've
played with this a bit).

My concern is that as we pop additional nodes running one of our
components, each of those will want to consume _n_ connections out of the
available 250 and eventually we are going to run out.

Looking through the code a bit today, I don't see how to use the Riak
connections outside of either RiakExternalLoadBalancer or RiakCluster.
Since RiakNode and RiakConnection are internal classes I don't see an easy
way to do this. Ideas?

Thanks in advance!

Mark Morga
